From 73ceb54e264576ac1c6db7a9c368a0337ceaf5f5 Mon Sep 17 00:00:00 2001
From: Joseph Fernandes <josferna@redhat.com>
Date: Wed, 20 Jan 2016 22:27:33 +0530
Subject: [PATCH 251/251] tier/dht : Default value for demote-freq, max files and mb

Default value for tier-demote-frequency is 3600 sec to avoid
frequent demotions.

Default value for tier-max-mb is 4000 mb

Default value for tier-max-files is 10000 files

Backport of http://review.gluster.org/13270/

>Change-Id: Ie60951c478a7462c425059699ab82511aa13fa0a
>BUG: 1300412
>Signed-off-by: Joseph Fernandes <josferna@redhat.com>

Change-Id: I040f038366aa9e4e1d7773d86cad6af74e68bc64
BUG: 1283961
Signed-off-by: Joseph Fernandes <josferna@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/66032
Reviewed-by: Dan Lambright <dlambrig@redhat.com>
Tested-by: Dan Lambright <dlambrig@redhat.com>
---
 xlators/cluster/dht/src/dht-shared.c            |    6 +++---
 xlators/cluster/dht/src/tier.c                  |    5 +++--
 xlators/mgmt/glusterd/src/glusterd-volume-set.c |    6 +++---
 3 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/xlators/cluster/dht/src/dht-shared.c b/xlators/cluster/dht/src/dht-shared.c
index 9d869e3..84f86bd 100644
--- a/xlators/cluster/dht/src/dht-shared.c
+++ b/xlators/cluster/dht/src/dht-shared.c
@@ -1008,7 +1008,7 @@ struct volume_options options[] = {
 
         { .key  = {"tier-demote-frequency"},
           .type = GF_OPTION_TYPE_INT,
-          .default_value = "120",
+          .default_value = "3600",
           .description = "Frequency to demote files to slow tier"
         },
 
@@ -1035,11 +1035,11 @@ struct volume_options options[] = {
         },
         { .key         = {"tier-max-mb"},
           .type = GF_OPTION_TYPE_INT,
-          .default_value = "1000",
+          .default_value = "4000",
         },
         { .key         = {"tier-max-files"},
           .type = GF_OPTION_TYPE_INT,
-          .default_value = "5000",
+          .default_value = "10000",
         },
         /* switch option */
         { .key  = {"pattern.switch.case"},
diff --git a/xlators/cluster/dht/src/tier.c b/xlators/cluster/dht/src/tier.c
index 0fe9c19..7c15b5a 100644
--- a/xlators/cluster/dht/src/tier.c
+++ b/xlators/cluster/dht/src/tier.c
@@ -1823,7 +1823,7 @@ tier_init (xlator_t *this)
                 freq = DEFAULT_TIER_MAX_MIGRATE_MB;
         }
 
-        defrag->tier_conf.max_migrate_bytes = freq * 1024 * 1024;
+        defrag->tier_conf.max_migrate_bytes = (uint64_t) freq * 1024 * 1024;
 
         ret = dict_get_int32 (this->options,
                               "tier-max-files", &freq);
@@ -2000,7 +2000,8 @@ tier_reconfigure (xlator_t *this, dict_t *options)
                 GF_OPTION_RECONF ("tier-max-mb",
                                   migrate_mb, options,
                                   int32, out);
-                defrag->tier_conf.max_migrate_bytes = migrate_mb*1024*1024;
+                defrag->tier_conf.max_migrate_bytes = (uint64_t) migrate_mb *
+                                                      1024 * 1024;
 
                 GF_OPTION_RECONF ("tier-max-files",
                                   defrag->tier_conf.max_migrate_files, options,
diff --git a/xlators/mgmt/glusterd/src/glusterd-volume-set.c b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
index 44f74a3..b854344 100644
--- a/xlators/mgmt/glusterd/src/glusterd-volume-set.c
+++ b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
@@ -2327,7 +2327,7 @@ struct volopt_map_entry glusterd_volopt_map[] = {
         },
         { .key         = "cluster.tier-demote-frequency",
           .voltype     = "cluster/tier",
-          .value       = "120",
+          .value       = "3600",
           .option      = "tier-demote-frequency",
           .op_version  = GD_OP_VERSION_3_7_0,
           .flags       = OPT_FLAG_CLIENT_OPT,
@@ -2371,7 +2371,7 @@ struct volopt_map_entry glusterd_volopt_map[] = {
         { .key         = "cluster.tier-max-mb",
           .voltype     = "cluster/tier",
           .option      = "tier-max-mb",
-          .value       = "10000",
+          .value       = "4000",
           .op_version  = GD_OP_VERSION_3_7_6,
           .flags       = OPT_FLAG_CLIENT_OPT,
           .validate_fn = validate_tier,
@@ -2381,7 +2381,7 @@ struct volopt_map_entry glusterd_volopt_map[] = {
         { .key         = "cluster.tier-max-files",
           .voltype     = "cluster/tier",
           .option      = "tier-max-files",
-          .value       = "50000",
+          .value       = "10000",
           .op_version  = GD_OP_VERSION_3_7_6,
           .flags       = OPT_FLAG_CLIENT_OPT,
           .validate_fn = validate_tier,
-- 
1.7.1

